src/components/Chart/constants.js

```diff
@@ -0,0 +1,104 @@
+import PropTypes from 'prop-types';
+import {
+  compose,
+  setDisplayName,
+  mapProps,
+  defaultProps,
+  setPropTypes,
+} from 'recompose';
+import C3Chart from 'react-c3js';
+
+import 'patternfly/dist/js/patternfly-settings';
+import 'patternfly/dist/js/patternfly-settings-charts';
+
+const patternfly = window.patternfly;
+const c3ChartDefaults = patternfly.c3ChartDefaults();
+const CHART_CONFIG = {
+  AREA_CHART: {
+    type: 'area',
+    displayName: 'AreaChart',
+    className: 'area-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultAreaConfig(),
+  },
+  BAR_CHART: {
+    type: 'bar',
+    displayName: 'BarChart',
+    className: 'bar-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultBarConfig(),
+  },
+  DONUT_CHART: {
+    type: 'donut',
+    displayName: 'DonutChart',
+    className: 'donut-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultDonutConfig(),
+  },
+  GROUPED_BAR_CHART: {
+    type: 'bar',
+    displayName: 'GroupedBarChart',
+    className: 'bar-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultGroupedBarConfig(),
+  },
+  LINE_CHART: {
+    type: 'line',
+    displayName: 'LineChart',
+    className: 'line-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultLineConfig(),
+  },
+  PIE_CHART: {
+    type: 'pie',
+    displayName: 'PieChart',
+    className: 'pie-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultPieConfig(),
+  },
+  SINGLE_AREA_CHART: {
+    type: 'area',
+    displayName: 'SingleAreaChart',
+    className: 'area-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultSingleAreaConfig(),
+  },
+  SINGLE_LINE_CHART: {
+    type: 'line',
+    displayName: 'SingleLineChart',
+    className: 'line-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultSingleLineConfig(),
+  },
+  SPARKLINE_CHART: {
+    type: 'area',
+    displayName: 'SparklineChart',
+    className: 'chart-pf-sparkline',
+    defaultConfig: c3ChartDefaults.getDefaultSparklineConfig(),
+  },
+  STACKED_BAR_CHART: {
+    type: 'bar',
+    displayName: 'StackedBarChart',
+    className: 'bar-chart-pf',
+    defaultConfig: c3ChartDefaults.getDefaultStackedBarConfig(),
+  },
+};
+
+const addDataType = props => {
+  if (props.data && !props.data.type) {
+    props.data.type = props.type;
+  }
+  return props;
+};
+
+const getComposer = name => {
+  const config = CHART_CONFIG[name];
+
+  return compose(
+    setDisplayName(config.displayName),
+    setPropTypes({
+      data: PropTypes.object.isRequired,
+      ...C3Chart.propTypes,
```

Comment:
nice!

Reply:
Yes... recompose is a welcome addition i.m.o.! I also noticed material-ui using it before...